docs/modules/ROOT/pages/configuration/global-filters.adoc

```diff
@@ -0,0 +1,95 @@
+= Configuring Global Filters on KubeArchive
+
+KubeArchive allows cluster owners to configure global filters. Those are applied
+to any KubeArchive namespace containing a `KubeArchiveConfig` custom resource.
+
+To configure the global filters create a `KubeArchiveConfig` in the
+"kubearchive" namespace. Because it is created on the "kubearchive" namespace
+these filters are considered global:
+
+[source,yaml]
+----
+---
+apiVersion: kubearchive.kubearchive.org/v1alpha1
+kind: KubeArchiveConfig
+metadata:
+  name: kubearchive
+  namespace: kubearchive
+spec:
+  resources:
+    - selector:
+        apiVersion: v1
+        kind: Pod
+      archiveOnDelete: "true"
+----
+
+== Interaction With Local Filters
+
+Global filters only work in namespaces containing a `KubeArchiveConfig` custom
+resource. These can be understood as "local" filters in opposition with global
+filters. Both global and local filters are combined and used for that specific
+namespace.
+
+For example the following `KubeArchiveConfig` on the namespace "my-team" KubeArchive
+archives jobs as long as they have `status.startTime` and deletes them when they
+have `status.completionTime`. However KubeArchive also archives pods when they are
+about to be deleted, as configured in the global filters in the previous section:
+
+[source,yaml]
+----
+---
+apiVersion: kubearchive.kubearchive.org/v1alpha1
+kind: KubeArchiveConfig
+metadata:
+  name: kubearchive
+  namespace: my-team
+spec:
+  resources:
+    - selector:
+        apiVersion: batch/v1
+        kind: Job
+      archiveWhen: has(status.startTime)
+      deleteWhen: has(status.completionTime)
+----
+
+For example the following `KubeArchiveConfig` on the namespace "production" KubeArchive
+only archives pods when they are about to be deleted, as configured in the global filters
+in the previous section:
+
+[source,yaml]
+----
+---
+apiVersion: kubearchive.kubearchive.org/v1alpha1
+kind: KubeArchiveConfig
+metadata:
+  name: kubearchive
+  namespace: production
+spec: {}
+----
+
+== Skipping Global Filters
+
+Global filters are enforced but cluster owners can configure them so users
+can choose to avoid them. The following `KubeArchiveConfig` allows
+users to specify the label "prevent-archive" in their pods so they are not
+archived:
```

Comment:
maybe I missed discussion about this, but this is also functionality that will need to be added to the sink

Reply:
I'm pretty sure this functionality is provided by the CEL expressions. If the global filters says "archive everything but things starting with "abc", you can prefix your thing with "abc" and it won't be archived. Does this make sense to you?

Reply:
> Global filters are enforced but cluster owners can configure them so users
can choose to avoid them. The following `KubeArchiveConfig` allows
users to specify the label "prevent-archive" in their pods so they are not
archived:

No, no, no. This is not permitted.

Reply:
> No, no, no. This is not permitted.

I'm not sure if you are joking :rofl:

We can't enforce what the cluster owner writes in their KubeArchiveConfig custom resources, so they can have any exceptions they want.

Reply:
>We can't enforce what the cluster owner writes in their KubeArchiveConfig custom resources, so they can have any exceptions they want.

Actually, we can. They can write it, but we don't have to use it.

See [this code](https://github.com/kubearchive/kubearchive/blob/main/cmd/operator/internal/controller/kubearchiveconfig_controller.go#L435-L441).

Even if we allowed the global configs to use it, there's no way for a user to specify `ApiServerSource` configuration specific to their namespace.

Reply:
I see. So this means that the specific example is wrong, as `kar.Selector.Selector` is not passed. However this is achievable with the CEL expressions. Is this right?

Reply:
> However this is achievable with the CEL expressions. Is this right?

What is achievable? The user overriding what is specified in the global filters?  No, I don't think it is.

Reply:
> What is achievable? The user overriding what is specified in the global filters? No, I don't think it is.

That is clear, users can't override global filters.

In a scenario where the global `archiveWhen` is:

```
!(has(metadata.labels) && 'prevent-archive' in metadata.labels)
```

If a user has a resource that contains the `prevent-archive` label, that resource does not get archived. This is not the user avoiding the global filters, but the cluster owners providing a hole in their filter so users can choose to skip it.

Reply:
Sure, that will work. The cluster admin is allowing it.

But the user isn't doing anything with their filters to allow it to happen.

I don't know why we have to document this specific use case.

Reply:
I believe that this information is useful for the cluster admins, in case they want to configure kubeArchive in a flexible way for the users and also for the users in case they want to start a conversation with the cluster admin and see if an exception could be provided.

Reply:
I see this as more of tip rather than a feature, but think pointing out stuff like this can be useful

Reply:
We can document this if you like. But we can't give tips on everything. We have to assume some knowledge of CEL expressions and how they can be used.  There's logic involved here, and there's no way around it.

You can also have things archived after a certain time interval has passed.  Or if the name begins with a certain prefix. Or if they have an annotation.  Or a particular status. There's a million different options.

 And while this is possible and your use case is plausible, it's not what we were asked for.  The request was to be able to set global filtering to *force* archiving regardless of what the user specifies.  What's implemented allows for both.

Reply:
> The request was to be able to set global filtering to force archiving regardless of what the user specifies

The thing is that when I read that global filtering "forces" archiving, I can't help to ask: but is there any way to avoid that or create exceptions?

My intention wasn't to document exactly one way of not applying the global filters for all resources of a kind but just letting know the reader that CEL expressions are flexible enough to generate exceptions for archiving a resource so the different namespaces can generate "non-archivable" resources if the global filter allows that.

If you think that giving an example is too much, I am ok with using just a sentence in a "TIP" section.

Reply:
After let it simmer for a bit, I will remove the block and merge without it. I there is a lot of questions around this we can add the block again to the documentation.

Reply:
closing this thread since the lines for this comment were removed